Osd message showtime

vierfuffzig opened the pull request:
adds parameter OSDn_MSG_TIME to set message item's show duration in seconds

tridge commented on libraries/AP_OSD/AP_OSD.h:
```diff
@@ -59,12 +59,15 @@ class AP_OSD_Screen {
     AP_Int8 enabled;
     AP_Int16 channel_min;
     AP_Int16 channel_max;
+    AP_Int8 msgtime;
```
it would be good to make the units clear in the name, so maybe msgtime_s ?

tridge commented on libraries/AP_OSD/AP_OSD.h:
```diff
@@ -59,12 +59,15 @@ class AP_OSD_Screen {
     AP_OSD *osd;
 
-    static const uint16_t message_show_time_ms = 20000;
+    static const uint16_t message_show_time_ms = 1000;
```
why do we need this variable now?

tridge commented on libraries/AP_OSD/AP_OSD.h:
```diff
@@ -59,12 +59,15 @@ class AP_OSD_Screen {
     AP_Int16 channel_max;
+    AP_Int8 msgtime;
+
+    AP_OSD_Screen(int8_t msgtime);
```
what is this change for?

tridge commented on libraries/AP_OSD/AP_OSD_Screen.cpp:
```diff
@@ -153,6 +153,13 @@ const AP_Param::GroupInfo AP_OSD_Screen::var_info[] = {
+    // @Description: Sets message duration seconds
+    // @Range: 1 20
+    // @User: Standard
+    AP_GROUPINFO("MSG_TIME", 28, AP_OSD_Screen, msgtime, 10),
```
this is adding it per-screen. Is that needed?

tridge commented on libraries/AP_OSD/AP_OSD_Screen.cpp:
```diff
@@ -422,9 +429,10 @@ void AP_OSD_Screen::draw_batused(uint8_t x, uint8_t y)
 void AP_OSD_Screen::draw_message(uint8_t x, uint8_t y)
 {
     AP_Notify * notify = AP_Notify::instance();
+    uint16_t duration = message_show_time_ms * msgtime;
```
I think you're just conventing from s to ms here? Just multiple by constant 1000 instead of using a variable

vierfuffzig commented:
thanks @tridge for helping a noob to figure things! 
moved from per-screen to general
deleted useless defs and variables
renamed variable respectively.

sh83 commented:
Looks good, thank you.

tridge commented:
I've squashed the commits and rebased
